Gems/Atom/RHI/Code/Source/RHI.Edit/Utils.cpp

```diff
@@ -155,6 +155,11 @@ namespace AZ
                 AZ_Error(ShaderPlatformInterfaceName, false, "%s", prependFileLoadResult.GetError().c_str());
                 return arguments.m_sourceFile;
             }
+            else if (strcmp(prependFileLoadResult.GetValue().end(), "\n") != 0)
+            {
+                // Add new line to prepend file if not present
+                prependFileLoadResult.GetValue().append("\n");
+            }
```

Comment:
This doesn't append a newline to the end of the string.
The `end()` function returns an iterator past the end of the string which is the null-terminator.

Instead the https://en.cppreference.com/w/cpp/string/basic_string/ends_with should be used instead
```suggestion
            else if (rependFileLoadResult.GetValue().ends_with("\n"))
            {
                // Add new line to prepend file if not present
                prependFileLoadResult.GetValue() += "\n";
            }
```

Reply:
Thanks for correcting this rookie mistake of mine :)